train.py

```diff
@@ -294,6 +294,29 @@ def buildCallbacks(params, model, dataset):
     return callbacks
 
 
+def save_random_states(write_path, user_seed=None):
+    import numpy.random
+    import random
+    import json
+
+    n = numpy.random.get_state()
+    p = random.getstate()
+
+    n = list(numpy.random.get_state())
+    n[1] = n[1].tolist()
+
+    p = list(numpy.random.get_state())
```

Comment:
lol no.

You accidentally overwrite `p` which is supposed to be the Python random state.

Reply:
Thanks for the spot!